src/states/client.rs

```diff
@@ -200,7 +210,40 @@ impl Client {
             return Ok(Transition::Stay);
         }
 
-        Ok(self.dispatch_routing_message(routing_msg.clone(), outbox))
+        Ok(self.dispatch_routing_message(routing_msg, outbox))
+    }
+
+    fn handle_direct_message(&mut self,
+                             direct_msg: DirectMessage,
+                             outbox: &mut EventBox)
+                             -> Result<Transition, RoutingError> {
+        use messages::DirectMessage::*;
+        match direct_msg {
```

Comment:
To avoid forgetting to consider this function in the case that a new variant is added to `DirectMessage`.  I'd be OK with your proposed change, but I guess a consistent approach would be good - seems in some cases we want the explicit list and others we're OK with matching on `_` or equivalent.